content/en/community/hackathons/sessions/bincompat/content/Work-Items_00.-Setup.md

```diff
@@ -1,25 +1,89 @@
-To easily setup, build and run Linux ELFs with [`app-elfloader`](https://github.com/unikraft/app-elfloader), best way is to use [the `scripts` repository](https://github.com/unikraft-upb/scripts).
-Clone [the `scripts` repository](https://github.com/unikraft-upb/scripts) on your machine to get started.
+To easily setup, build and run Linux ELFs with [`app-elfloader`](https://github.com/unikraft/app-elfloader), best way is to use [the `run-app-elfloader` repository](https://github.com/unikraft/run-app-elfloader).
+Clone [the `run-app-elfloader` repository](https://github.com/unikraft/run-app-elfloader) on your machine to get started.
+Along with the `run-app-elfloader` repository, we have collected some already compiler applications that you can use to test the `elfloader`.
+Those are located in the [`static-pie-apps`](https://github.com/unikraft/static-pie-apps/) and [`dynamic-apps`](https://github.com/unikraft/dynamic-apps/) repositories.
+It's just a bunch of already compiled applications, so you don't have to spend time compiling stuff, there is nothing Unikraft-related in those repositories.
 
 ```console
-$ git clone https://github.com/unikraft-upb/scripts
+$ git clone https://github.com/unikraft/run-app-elfloader
+$ git clone https://github.com/unikraft/static-pie-apps
+$ git clone https://github.com/unikraft/dynamic-apps
 
-$ cd scripts/
+$ cd run-app-elfloader/
 
-$ cd make-based/app-elfloader/
+$ ./run_app.sh helloworld
 
-$ ./do.sh setup
+SeaBIOS (version rel-1.16.2-0-gea1b7a073390-prebuilt.qemu.org)
+Booting from ROM..TEST nofollow
+Powered by
+o.   .o       _ _               __ _
+Oo   Oo  ___ (_) | __ __  __ _ ' _) :_
+oO   oO ' _ `| | |/ /  _)' _` | |_|  _)
+oOo oOO| | | | |   (| | | (_) |  _) :_
+ OoOoO ._, ._:_:_,\_._,  .__,_:_, \___)
+                  Atlas 0.13.1~d20aa7cb
+[...]
+brk(NULL) = va:0x400200000
+brk(va:0x400221000) = va:0x400221000
+Hello, World!
+write(fd:1, "Hello, World!\x0A", 14) = 14
+qemu-system-x86_64: terminating on signal 2
+```
+
+This will run a dynamically linked `helloworld` application.
+You will see a lot of output because, by default, the `run_app.sh` script uses a prebuilt version of `app-elfloader`, which has `strace`-like output enabled, for debugging purposes.
+You can see a list of all existing available application that you can run in binary compatibility mode by running `./run_app.sh` with no arguments.
+**Note: If you are running an application that requires networking support, you will need to run it with `sudo`.***
+
+If you want more control over the application, you can use the `run.sh` script.
```

Comment:
This is too much. I would move this to section 1.5 (i.e. between current sections 1 and 2, with a name such as "Behind the Scenes"). The "storyline" would be like:

* `00. Setup`: Quick setting up of binary compatibility. Have everything ready, run helloworld.

* `01. Run Binary Applications`: Run all the applications for binary compatibility. Use `run_app.sh`.

* `02: Flavors of ELF Loader`: Detail existing pre-built version of `elfloader` and updating them via the `defaults` file. Continue using `run_app.sh`.

* `03. Behind the Scenes`: Introduce the `run.sh` script, run commands using `run.sh`. Show filesystem information in the `dynamic-apps` / `static-apps` repository. The contents detailing `run.sh` from this setup section together with contents from current 03 section will end up here.

* `04. Run New Applications`: Use `extract.sh` script or a manual approach to create a filesystem for a simple CLI application already installed (or that needs to be installed) in the system (`tree` or `zip` or `unzip` or anything dealing with the filesystem - it's easier; for processes, you require `procfs` support and that's more difficult).

* `05. Build and Run Custom Applications`: Provide a sample source code or link to a repository where they would build and run both static and dynamic apps. I would give two subtasks. A simple one, from some toy repository (something like [my `snippets` repository](https://github.com/razvand/snippets)), with an existing `Makefile` with rules to build a static and dynamic application. And then to a practical one (I would go for a Go or Rust repository) that they clone, build, run locally and the run via the `elfloader`.
  The thing is that build the ELF Loader is not that relevant. But running different applications, building them, that's more practical.

* `06: Build app-elfloader`: First show a full command to run `helloworld` and `nginx`. Show the usefulness of the `run_app.sh` and the `run.sh` script., that hides the complexity of using `qemu-system-x86_64`. Then proceed to build `app-elfloader`.

* `07: Debugging Binary Compatibility`: This is a bonus item and can be left out. I would ask them to run a process/memory/user-related CLI executable (`free`, `id` etc.). And see that it requires a certain filesystem (probably files in `/etc/` or `/proc/`) and ask them to create the proper filesystem for it to run.